app/models/unit.rb

```diff
@@ -363,13 +363,23 @@ def unenrol_users_from_csv(file)
 
         project_participant = project_participant.first
 
-        user_project = projects.where(user_id: project_participant.id).first
```

Comment:
Looks like this change has not gone across. UnitRole is for staff only, so update to use this in the new change.